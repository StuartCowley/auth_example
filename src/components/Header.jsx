import "../styles/header.css"
import { Link } from "react-router-dom"

import { useContext } from "react"
import AuthContext from "../utils/AuthContext"

const Header = () => {
    const { user } = useContext(AuthContext);

    return (
        <nav className="header">
            {user?.username}
            <Link to="/">Home</Link>
            <Link to="/account">Account</Link>
            <Link to="/login">Login</Link>
        </nav>
    )
}

export default Header