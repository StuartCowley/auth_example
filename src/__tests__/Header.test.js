import { render, screen } from "@testing-library/react";
import { BrowserRouter as Router } from "react-router-dom";
import renderer from "react-test-renderer";
import Header from "../components/Header";

describe("Login form component", () => {
    it("matches snapshot", () => {
        const generated = renderer
            .create(
                <Router>
                    <Header />
                </Router>
            ).toJSON();

        expect(generated).toMatchSnapshot();
    });

    it("all links render as expected", () => {
        render(
            <Router>
                <Header />
            </Router>
        );
        const homeLink = screen.getByRole("link", { name: "Home" });
        const loginLink = screen.getByRole("link", { name: "Login" });
        const accountLink = screen.getByText("Account");

        expect(homeLink).toBeInTheDocument;
        expect(loginLink).toBeInTheDocument;
        expect(accountLink).toBeInTheDocument;
    });
});