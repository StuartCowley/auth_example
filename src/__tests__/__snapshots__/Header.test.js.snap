// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login form component matches snapshot 1`] = `
<nav
  className="header"
>
  <a
    href="/"
    onClick={[Function]}
  >
    Home
  </a>
  <div
    className="header__nav-link"
    onClick={[Function]}
  >
    Account
  </div>
  <a
    href="/login"
    onClick={[Function]}
  >
    Login
  </a>
</nav>
`;
